lib/mongoid/extensions/big_decimal.rb

```diff
@@ -58,7 +68,17 @@ def demongoize(object)
         #
         # @return [ String, nil ] A String representing the object or nil.
         def mongoize(object)
-          object && object.numeric? ? object.to_s : nil
+          if object
+            if Mongoid.map_big_decimal_to_decimal128
+              if object.is_a?(BigDecimal)
+                BSON::Decimal128.new(object)
+              elsif object.numeric?
+                BSON::Decimal128.new(object.to_s)
+              end
```

Comment:
... and I would like to have an `else` here returning `object`, same as with the long comment above.

Reply:
At a minimum the `object` must be mongoized, so return `object.mongoize`

Reply:
Won't that cause an infinite loop @johnnyshields ?

Reply:
No.

Reply:
Ah I see. For my curiosity, can you point out an example where this would cause a change in functionality?

Reply:
Probably none in practice, but the docs state that SomeObject.mongoize(...) class method should be a catch-all for all possible types, even ones that do not cleanly map to SomeObject.

Reply:
Okay, I'm willing to change that. Done.